Patch-Source: https://github.com/libass/libass/commit/e5f4f0d4b103a3119bafa848e0811f4e7d5cd190
https://github.com/libass/libass/releases/tag/0.17.1
--
From e5f4f0d4b103a3119bafa848e0811f4e7d5cd190 Mon Sep 17 00:00:00 2001
From: Oleg Oshmyan <chortos@inbox.lv>
Date: Sat, 21 Jan 2023 16:59:14 +0200
Subject: [PATCH] WHOLE_TEXT_LAYOUT bidi: fix buffer overread on soft-wrapped
 events

See commit cc54eea644db29bc64bdaa9305576d4d18cae9ed, which introduced this:
we have a base direction for each *paragraph* delimited by an explicit
line break (allocated and filled in ass_shaper_shape), not for each
display line (which the current ass_shaper_reorder code attempts to read).

Fixes https://github.com/libass/libass/issues/672.

Cherry-picked from: c1daedea0e829a7e90389818d2bb662fd91d0be7
---
 libass/ass_shaper.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libass/ass_shaper.c b/libass/ass_shaper.c
index db61963cc..77cf90906 100644
--- a/libass/ass_shaper.c
+++ b/libass/ass_shaper.c
@@ -1102,7 +1102,7 @@ FriBidiStrIndex *ass_shaper_reorder(ASS_Shaper *shaper, TextInfo *text_info)
                 return NULL;
 
             last_break = i + 1;
-            if (shaper->whole_text_layout)
+            if (shaper->whole_text_layout && glyphs[i].symbol == '\n')
                 pdir++;
         }
     }
